incubator/hnc/pkg/controllers/hierarchy_controller.go

```diff
@@ -477,6 +482,11 @@ func (r *HierarchyReconciler) writeHierarchy(ctx context.Context, log logr.Logge
 		return false, nil
 	}
 
+	if !ex[inst.ObjectMeta.Namespace] {
+		atomic.AddInt32(&apiCall, 1)
+		atomic.AddInt32(&hcWriteHC, 1)
```

Comment:
Not sure we need _both_ `apiCall` and `hcWriteHC` (which should just be `hcWrite`). We can aggregate somewhere else, not at the call site.

Also, as a TODO, I wonder if we can replace this completely with the controller-runtime Prometheus metrics?

https://github.com/kubernetes-sigs/controller-runtime/blob/838c86fe0af6cd645301059505a61742de01ea2c/pkg/metrics/client_go_adapter.go#L35